docs/_components/form-fields.md

```diff
@@ -1,22 +1,571 @@
+
+# Form field groups
+
+{% capture example %}
+<fieldset class="usa-fieldset">
+  <legend class="usa-legend">Full name</legend>
+
+  <label for="c985" class="usa-label">First name</label>
+  <input id="c985" type="text" class="usa-input">
+
+  <label for="a2d8" class="usa-label">Last name</label>
+  <input id="a2d8" type="text" class="usa-input">
+</fieldset>
+{% endcapture %}
+{% include helpers/code-example.html code=example %}
+
+# Form validation errors
+
+{% capture example %}
+<label for="e9a4" class="usa-label">Field label</label>
+<input id="e9a4" type="text" class="usa-input usa-input-error">
+<span class="usa-input-error-message" role="alert">Error message text.</span>
+{% endcapture %}
+{% include helpers/code-example.html code=example %}
+
+## Other input types
+
+Follow the same pattern of applying `.usa-input-error` to the effected input and append the error message in the appropriate span.
+
+<div class="usa-accordion usa-accordion-bordered">
+  <h3 class="usa-accordion-heading">
+    <button class="usa-accordion-button" aria-controls="errored-textarea">
+      Textarea (multiline) input
+    </button>
+  </h3>
+  <div id="errored-textarea" class="usa-accordion-container">
+    <div class="usa-accordion-content">
+{% capture example %}
+<label for="a37c" class="usa-label">Textarea (multiline) input</label>
+<textarea id="a37c" type="text" class="usa-textarea usa-input-error"></textarea>
+<span class="usa-input-error-message" role="alert">Error message text.</span>
+{% endcapture %}
+{% include helpers/code-example.html code=example %}
+    </div>
+    <button class="usa-accordion-close-button">Close</button>
+  </div>
+
+  <h3 class="usa-accordion-heading">
+    <button class="usa-accordion-button" aria-controls="errored-dates">
+      Dates
+    </button>
+  </h3>
+  <div id="errored-dates" class="usa-accordion-container">
+    <div class="usa-accordion-content">
+{% capture example %}
+<fieldset class="usa-fieldset">
+  <legend class="usa-legend">Date of birth</legend>
+  <div class="usa-memorable-date">
+    <div class="usa-form-group usa-form-group-month">
+      <label for="e30e" class="usa-label">Month</label>
+      <input id="e30e" class="usa-input usa-input-inline usa-input-error" aria-describedby="f9e5" type="number" min="1" max="12">
+    </div>
+    <div class="usa-form-group usa-form-group-day">
+      <label for="b857" class="usa-label">Day</label>
+      <input id="b857" class="usa-input usa-input-inline usa-input-error" aria-describedby="f9e5" type="number" min="1" max="31">
+    </div>
+    <div class="usa-form-group usa-form-group-year">
+      <label for="a5dc" class="usa-label">Year</label>
+      <input id="a5dc" class="usa-input usa-input-inline usa-input-error" aria-describedby="f9e5" type="number" min="1900" max="2000">
+    </div>
+  </div>
+  <span class="usa-input-error-message" role="alert">Error message text.</span>
+  <span class="usa-form-hint" id="f9e5">Example: 04 28 1986</span>
```

Comment:
Also thinking that the error should appear on a line by itself so the hint necessary to validate the field is still aligned with the fields.

I'll say the error should come after the hint but open to suggestions as I don't really have a good example to go off of.

Reply:
I’ll go with after, I also think it seems to make more sense logically (the error state doesn’t break apart the existing control, which I imagine contains the hint). 

Here’s both options visually, just for reference. 

| Before hint | After hint |
|------------|----------|
| ![Screenshot of errored date field with error message showing before hint text](https://user-images.githubusercontent.com/14930/51941889-a6882780-23e3-11e9-9205-845eda5e679e.png) | ![Screenshot of errored date field with error message showing after hint text](https://user-images.githubusercontent.com/14930/51941768-65901300-23e3-11e9-9186-9fc798384545.png) |

Reply:
👍  on after